src/solver/optimisation/adequacy_patch.cpp

```diff
@@ -67,72 +77,239 @@ LinkCapacityForAdequacyPatchFirstStep SetNTCForAdequacyFirstStepOriginNodeInside
+        densNew = Math::Max(0.0, ensInit + netPositionInit + flowsNode1toNodeA);
+    else
+        densNew = Math::Max(0.0, ensInit + netPositionInit);
+
+    return std::make_pair(netPositionInit, densNew);
 }
 
-void setBoundsNoAdqPatch(double& Xmax,
-                         double& Xmin,
-                         VALEURS_DE_NTC_ET_RESISTANCES* ValeursDeNTC,
-                         const int Interco)
+void addArray(std::vector<double>& A, const double* B)
 {
-    Xmax = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
-    Xmin = -(ValeursDeNTC->ValeurDeNTCExtremiteVersOrigine[Interco]);
+    for (uint i = 0; i < A.size(); ++i)
+        A[i] += B[i];
 }
-} // namespace AdequacyPatch
+
+} // end namespace Antares
 } // end namespace Data
-} // namespace Antares
+} // end namespace AdequacyPatch
+
+void HOURLY_CSR_PROBLEM::calculateCsrParameters()
+{
+    double netPositionInit;
+    double densNew;
+    double ensInit;
+    double spillageInit;
+    int hour = hourInWeekTriggeredCsr;
+
+    for (int Area = 0; Area < pWeeklyProblemBelongedTo->NombreDePays; Area++)
+    {
+        if (pWeeklyProblemBelongedTo->adequacyPatchRuntimeData.areaMode[Area]
+            == physicalAreaInsideAdqPatch)
+        {
+            std::tie(netPositionInit, densNew)
+              = calculateAreaFlowBalance(pWeeklyProblemBelongedTo, Area, hour);
+
+            ensInit = pWeeklyProblemBelongedTo->ResultatsHoraires[Area]
+                        ->ValeursHorairesDeDefaillancePositive[hour];
+            spillageInit = pWeeklyProblemBelongedTo->ResultatsHoraires[Area]
+                             ->ValeursHorairesDeDefaillanceNegative[hour];
+
+            densNewValues[Area] = densNew;
+            rhsAreaBalanceValues[Area] = ensInit + netPositionInit - spillageInit;
+        }
+    }
+    return;
+}
+
+void HOURLY_CSR_PROBLEM::resetProblem()
+{
+    OPT_LiberationProblemesSimplexe(pWeeklyProblemBelongedTo);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemVariables()
+{
+    OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo,
+                                                                         *this);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemConstraintsLHS()
+{
+    OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::setVariableBounds()
+{
+    OPT_InitialiserLesBornesDesVariablesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemConstraintsRHS()
+{
+    OPT_InitialiserLeSecondMembreDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::setProblemCost()
+{
+    OPT_InitialiserLesCoutsQuadratiques_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::solveProblem(uint week, int year)
+{
+    ADQ_PATCH_CSR(pWeeklyProblemBelongedTo->ProblemeAResoudre, *this, week, year);
+}
+
+void HOURLY_CSR_PROBLEM::run(uint week, int year)
```

Comment:
Many methods of **HOURLY_CSR_PROBLEM** are about building and running a CSR quadratic problem.
We should have a class **csrQuadraticProblem**, and **HOURLY_CSR_PROBLEM::run(...)** should run an instance of it.
In code word, we should have something like : 
```C++
void HOURLY_CSR_PROBLEM::run(uint week, int year)
{
    computeCSRquadraticProblemParameters();

    csrQuadraticProblem csr_quad_pb(hourInWeek,
                                    densNewValues, 
                                    rhsAreaBalanceValues); // and maybe other args
    csr_quad_pb.buildVariables();
    csr_quad_pb.buildProblemConstraints();
    ....
    csr_quad_pb.solve(week, year);
}
```

Reply:
This would make HOURLY_CSR_PROBLEM just a wrapper of the csrQuadraticProblem class adding complexity, I don't see the point